pkgs/racket-doc/scribblings/reference/hashes.scrbl

```diff
@@ -986,4 +986,115 @@ one for the result.
 
 @history[#:added "7.9.0.1"]}
 
+@defproc[(hash-filter [ht hash?] [pred (-> any/c any/c boolean?)])
+         hash?]{
+
+Filters the @racket[hash?] @racket[ht] based on a predicate @racket[pred] applied to both its keys and values.
+This function constructs a new hash table that includes only those key-value pairs
+from the input @racket[ht] for which the predicate @racket[pred] returns true when
+applied simultaneously to the keys and values of @racket[ht]. The output hash table
+retains the mutability and the key comparison predicate (e.g., @racket[eqv?],
+@racket[equal-always?], @racket[equal?]) of the input hash table @racket[ht], ensuring that
+the structural and operational properties of the original hash are preserved
+in the output. This allows for nuanced filtering where the inclusion of a key-value pair
+depends on both the key and value meeting the criteria defined by @racket[pred].
+
+@examples[
+  #:eval the-eval
+  ;; Filtering key-value pairs where the key is less than 3 and value is even
```

Comment:
Oh, these comments won't be rendered, btw. You need to use `code:comment`. Not sure how much you care about that.